src/Common/src/System/SR.Core.cs

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Internal.Runtime.Augments;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace System
+{
+    internal static partial class SR
+    {
+        //
+        // Lower level libraries like System.Private.CoreLib, System.Private.Uri and other System.Private.* libraries.
+        // cannot depend on System.Resources.ResourceManagere directly, so we'll call Windows Runtime ResourceManager 
+        // directly using Internal.Runtime.Augments. 
+        // For other assemblies, we use System.Resources.ResourceManager to do the resources lookup. it is important to 
+        // not have such assemblies depend on internal contratcs as we can decide to make these assemblies portable.
+        //
+
+        private static Object s_resourceMap;
+        private const string MoreInfoLink = @". For more information, visit http://go.microsoft.com/fwlink/?LinkId=623485";
+
+        private static Object ResourceMap
+        {
+            get
+            {
+                if (SR.s_resourceMap == null)
+                {
+                    SR.s_resourceMap = WinRTInterop.Callbacks.GetResourceMap(SR.s_resourcesName);
+                }
+                return SR.s_resourceMap;
+            }
```

Comment:
Nit: this could be simplified as follows:

```c#
return s_resourceMap = s_resourceMap ?? WinRTInterop.Callbacks.GetResourceMap(s_resourcesName);
```